examples/demo-app/src/cloud-providers/carto/carto-provider.js

```diff
@@ -0,0 +1,424 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {OAuthApp} from '@carto/toolkit';
+import Console from 'global/console';
+import CartoIcon from './carto-icon';
+import {formatCsv} from 'processors/data-processor';
+import {Provider} from 'kepler.gl/cloud-providers';
+
+const NAME = 'carto';
+const DISPLAY_NAME = 'CARTO';
+const NAMESPACE = 'keplergl';
+const DOMAIN = 'carto.com';
+const PRIVATE_STORAGE_ENABLED = true;
+const SHARING_ENABLED = true;
+
+export default class CartoProvider extends Provider {
+  constructor(clientId) {
+    super({name: NAME, displayName: DISPLAY_NAME, icon: CartoIcon});
+
+    this.clientId = clientId;
+    this.thumbnail = {width: 300, height: 200};
+    this.currentMap = null;
+
+    this._folderLink = `https://{user}.${DOMAIN}/dashboard/maps/external`;
+
+    // Initialize CARTO API
+    this._carto = new OAuthApp(
+      {
+        authorization: `https://${DOMAIN}/oauth2`,
+        scopes: 'schemas:c'
+      },
+      {
+        serverUrlTemplate: `https://{user}.${DOMAIN}/`,
+        namespace: NAMESPACE
+      }
+    );
+
+    this._carto.setClientID(clientId);
+
+    if (this.getAccessToken()) {
+      this.login();
+    }
+  }
+
+  /**
+   * The CARTO toolkit library takes care of the login process.
+   */
+  login(onCloudLoginSuccess) {
+    try {
+      this._carto.login().then(() => {
+        onCloudLoginSuccess && onCloudLoginSuccess(this.name);
+      });
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  logout(onCloudLogoutSuccess) {
+    try {
+      this._carto.oauth.clear();
+      this._carto.oauth._carto.sync();
+      onCloudLogoutSuccess();
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  isEnabled() {
+    return this.clientId !== null;
+  }
+
+  isConnected() {
```

Comment:
i don't think `isConnected` is used anymore.  in the UI we always call `const isConnected = Boolean(cloudProvider.getAccessToken()`

Reply:
👍